core/src/main/scala/io/projectglow/sql/expressions/ImputeMean.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 The Glow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.projectglow.sql.expressions
+
+import io.projectglow.sql.util.RewriteAfterResolution
+
+import org.apache.spark.sql.SQLUtils
+import org.apache.spark.sql.catalyst.expressions.aggregate.Average
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.util.ArrayData
+import org.apache.spark.sql.types.{ArrayType, DataType, DoubleType, LongType, StringType, StructType}
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * Imputes the missing values of an array using the mean of the non-missing values. Values that are NaN, null or
+ * equal to the missing value parameter are not included in the aggregation, and are substituted with the mean.
+ *
+ * If the missing value is not provided, the parameter defaults to -1.
+ */
+case class ImputeMean(array: Expression, missingValue: Expression) extends RewriteAfterResolution {
```

Comment:
What happens if the array element is not numeric? Or if it's numeric but not double? Might need to `ImplicitCastInputTypes` to doubles.